src/west/cmd/project.py

```diff
@@ -124,22 +124,38 @@ def __init__(self):
         super().__init__(
             'branch',
             _wrap('''
-            Create topic branch.
+            Create a branch or list branches, in multiple projects.
 
             Creates a branch in each of the specified projects (default: all
-            cloned projects).
+            cloned projects). The new branches are set to track '{}'.
 
-            The new branches are set to track '{}'.
+            With no arguments, lists all local branches along with the
+            repositories they appear in.
 
             '''.format(_MANIFEST_REV_BRANCH) + _MANIFEST_REV_HELP))
 
     def do_add_parser(self, parser_adder):
-        return _add_parser(parser_adder, self, _branch_arg, _project_list_arg)
+        return _add_parser(parser_adder, self, _opt_branch_arg,
+                           _project_list_arg)
 
     def do_run(self, args, user_args):
-        for project in _projects(args):
-            if _cloned(project):
+        projects = [project for project in _projects(args) if _cloned(project)]
+
+        if args.branch:
+            # Create a branch in the specified projects
+            for project in projects:
                 _create_branch(project, args.branch)
+        else:
+            # No arguments. List local branches from all projects along with
+            # the projects they appear in.
+
+            branch2projs = collections.defaultdict(list)
+            for project in projects:
+                for branch in _branches(project):
+                    branch2projs[branch].append(project.name)
+
+            for branch, projs in sorted(branch2projs.items()):
+                log.inf('{:18} {}'.format(branch, " ".join(projs)))
```

Comment:
Would you mind comma-separating the projects? I know we don't really support paths with whitespaces in Zephyr now, but it'd be nice not to rely on that.

Reply:
Sure, will do. Was thinking space-separation might be nicer for scripts and the like, but that's kinda obscure...

Reply:
Except if you have spaces in repository names at least... seems like a recipe for pain.

Reply:
Fair point -- let's keep this as is, then?